db/compaction/compaction_job.cc

```diff
@@ -1192,6 +1192,30 @@ void CompactionJob::ProcessKeyValueCompaction(SubcompactionState* sub_compact) {
       status = blob_file_builder->Finish();
     }
 
+#ifndef ROCKSDB_LITE
+    auto sfm =
```

Comment:
Same question for compaction, but didn't verify the details.

Reply:
Good point. I wonder why not also delete output file from a failed compaction immediately. This file may very likely be corrupted or inaccessible.

Reply:
> Good point. I wonder why not also delete output file from a failed compaction immediately. This file may very likely be corrupted or inaccessible.

Right, and I do wonder why it's not handled uniformly across flush/compaction. On the other hand, I think it's pretty harmless since the output of failed compactions does not get installed (it's just a matter of when trash gets cleaned up).

Reply:
@riversand963 @akankshamahajan15  In terms of reporting files to `SstFileManager`, I did some more digging and now I actually think what we do for compactions (i.e. reporting new files as we go) makes a lot of sense. Calling `OnAddFile` results in `total_file_size_` getting incremented in `SstFileManagerImpl`, which is then used for checking the "max allowed space reached" condition as well as allowing further compactions. Considering that compactions can run for a long time and generate a lot of data, I think doing it this way is better b/c it keeps `total_file_size_` more up-to-date.

In fact, we might want to report blob files as they are generated (or deleted) during flush too for the same reason. (Note that a flush job only generates a single SST; one way to look at the current code which only notifies `sfm` upon success is treating it as an optimization that eliminates a pair of `OnAddFile` / `OnDeleteFile` in failure cases.) Based on this, the right place to notify `sfm` would probably be in `BlobFileBuilder`.

Reply:
@ltamasi  Thanks for looking into it. It makes sense. I will notify the sfm in BlobFileBuilder.